cluster-autoscaler/Dockerfile.amd64

```diff
@@ -11,9 +11,17 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-ARG BASEIMAGE=gcr.io/distroless/static:nonroot-amd64
-FROM $BASEIMAGE
+FROM golang:1.23 AS builder
+
+WORKDIR /workspace
+
+COPY . .
+
+ARG ARCH amd64
+
+RUN GOOS=linux GOARCH=$ARCH go build -o cluster-autoscaler-$ARCH
```

Comment:
Let's add `CGO_ENABLED=0` here to match the Makefile? I think this can probably be hardcoded directly?

(at least that's what we do in https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/admission-controller/Dockerfile)

Reply:
went with the hardcoded option